### PR TITLE
clean up the mess in EB_icc.make_module_req_guess + other style fixesin icc/ifort easyblocks

### PR DESCRIPTION
**Pull request by boegel**
for https://github.com/hpcugent/easybuild-easyblocks/pull/756

@wpoely86: ready for thorough testing with this included (by me ;-))

### CODE REVIEW

File: easybuild/easyblocks/i/icc.py [hunk 1]
```diff
@@ -60,13 +61,11 @@ class EB_icc(IntelBase):
         - will fail for all older versions (due to newer silent installer)
     """
 
-    def configure_step(self):
-        """have fallback for components for > 2016 versions"""
+    def __init__(self, *args, **kwargs):
+        """Constructor, initialize class variables."""
+        super(EB_icc, self).__init__(*args, **kwargs)
 
-        if LooseVersion(self.version) >= LooseVersion('2016') and not self.cfg['components']:
```
Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py:66]: we don't really know what will happen with older version if you specify `components`? Maybe add a warning about?

File: easybuild/easyblocks/i/icc.py [hunk 2]
```diff
@@ -89,127 +88,124 @@ def install_step(self):
...
+        """
         prefix = None
-        tbbgccversion = get_tbb_gccprefix()
+
+        # guesses per environment variables
+        # some of these paths only apply to certain versions, but that doesn't really matter
+        # existence of paths is checked by module generator before 'prepend-paths' statements are included
+        guesses = {
+            'CLASSPATH': ['daal/lib/daal.jar'],
+            'CPATH': ['daal/include', 'include', 'ipp/include', 'mkl/include', 'tbb/include'],
```
Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py]: put `include` first

File: easybuild/easyblocks/i/icc.py [hunk 3]
```diff
@@ -89,127 +88,124 @@ def install_step(self):
...
+                'tbb/bin/intel64',
+            ])
+
+            # in the end we set 'LIBRARY_PATH' equal to 'LD_LIBRARY_PATH'
+            guesses['LD_LIBRARY_PATH'].extend([
+                'compiler/lib/intel64',
+                'debugger/ipt/intel64/lib',
+                'lib/intel64',
+                'ipp/lib/intel64',
+                'lib/intel64',
```
Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py]: duplicate

Comment (boegel) [inline easybuild/easyblocks/i/icc.py]: oops, thx

File: easybuild/easyblocks/i/icc.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/i/icc.py]: according to which alphabet? :)

File: easybuild/easyblocks/i/icc.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/i/icc.py:66]: As long as you pick the right set of components, you should be fine?

Trying to specify components shouldn't be a reason to raise a red flag imho, even for older versions (we know that the 2015 versions are also too fat by default, for example).

File: easybuild/easyblocks/i/icc.py (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py]: these shouldn't be alphabetically. `include` should come first (first used in the path), so it should be last?

Comment (boegel) [inline easybuild/easyblocks/i/icc.py]: why should it be first?
if a particular header file is available through several locations, it better be the same file?

besides, I just realised that `$CPATH` was never set before for the Intel compilers, and that hasn't been an issue, so I wonder what will happen if we do start prepending to it...

Comment (wpoely86) [inline easybuild/easyblocks/i/icc.py]: It shouldn't break anything. But I still believe `include` should be first and take preference before the others.

**Pull request merged**